Web/utils.py

```diff
@@ -69,23 +66,29 @@ def clean_process(self):
         if self.heartbeat and self.heartbeat.isAlive():
             self.heartbeat.stop = True
             self.heartbeat = None
-        print("Closed all threads")
+        logger.info("Closed all threads")
 
     def check_alive(self):
         if not self.process or self.process.poll() or self.stopped:
-            print("DEAD")
-            self.create()
+            logger.info("Java process went down")
+            return False
+        return True
 
     def stdin(self):
-        self.check_alive()
+        if not self.check_alive():
```

Comment:
Move all of this into a decorator and annotate every method using this with that decorator